src/wallet/wallet.cpp

```diff
@@ -1194,9 +1208,16 @@ void CWallet::ResendWalletTransactions(bool fForce)
             if (CheckTransaction(wtx)) {
                 wtx.RelayWalletTransaction(txdb);
             } else {
-                LogPrintf("ResendWalletTransactions() : CheckTransaction failed for transaction %s", wtx.GetHash().ToString());
+                to_be_erased.push_back(wtx);
             }
         }
+
+        for (const auto& wtx : to_be_erased) {
+            LogPrintf("%s: Erasing stale transaction %s.", __func__, wtx.GetHash().ToString());
+            EraseFromWallet(wtx.GetHash());
+            mempool.remove((CTransaction)wtx);
+            NotifyTransactionChanged(this, wtx.GetHash(), CT_DELETED);
+        }
```

Comment:
This is why the MRC requests are being deleted from the wallet. MRC requests that are CONFIRMED (i.e. make it into a block) should not be deleted, because they are valid and historical.